src/index.js

```diff
@@ -9,87 +9,93 @@
+    });
+
+    child.on('disconnect', (code, signal) => {
+      console.log(`child process disconnected from IPC channel with (${code} and ${signal})`);
+      // Killing child just in case it has not yet died.
+      child.kill(); // SIGTERM
+    });
+
+    child.on('exit', (code, signal) => {
+      console.log(`child process exited with code ${code} or due to signal ${signal}.`);
+      // Try to exit with the same signal as the child process.
+      process.exit(code || 127); // Have a default to indicate it was not a normal termination
+    });
+
+    // Pipe wallet events to child process
+    notificationBus.on(EVENTBUS_EVENT_NAME, data => {
+      if (child.killed || !child.connected) {
+        // The child has been lost, cannot send to it
+        return;
+      }
+      // Sending event data
+      child.send(data);
+    });
+  }
+
+  // Start main process
+
+  const app = createApp(config);
+
+  // Logging relevant variables on the console
+  console.log('Starting Hathor Wallet...', {
+    wallet: version,
+    version: process.version,
+    platform: process.platform,
+    pid: process.pid,
   });
 
-  child.on('exit', (code, signal) => {
-    console.log(`child process exited with code ${code} or due to signal ${signal}.`);
-    // Try to exit with the same signal as the child process.
-    process.exit(code || 127); // Have a default to indicate it was not a normal termination
+  console.log('Configuration...', {
+    network: config.network,
+    server: config.server,
+    txMiningUrl: hathorLibConfig.getTxMiningUrl(),
+    tokenUid: config.tokenUid,
+    apiKey: config.http_api_key,
+    gapLimit: config.gapLimit,
+    connectionTimeout: config.connectionTimeout,
+    enabled_plugins: config.enabled_plugins,
   });
 
-  // Pipe wallet events to child process
-  notificationBus.on(EVENTBUS_EVENT_NAME, data => {
-    if (child.killed || !child.connected) {
-      // The child has been lost, cannot send to it
-      return;
-    }
-    // Sending event data
-    child.send(data);
+  app.listen(config.http_port, config.http_bind_address, () => {
+    console.log(`Listening on ${config.http_bind_address}:${config.http_port}...`);
   });
 }
 
-// Start main process
-
-const app = createApp();
-
-// Logging relevant variables on the console
-console.log('Starting Hathor Wallet...', {
-  wallet: version,
-  version: process.version,
-  platform: process.platform,
-  pid: process.pid,
-});
-
-console.log('Configuration...', {
-  network: config.network,
-  server: config.server,
-  txMiningUrl: hathorLibConfig.getTxMiningUrl(),
-  tokenUid: config.tokenUid,
-  apiKey: config.http_api_key,
-  gapLimit: config.gapLimit,
-  connectionTimeout: config.connectionTimeout,
-  enabled_plugins: config.enabled_plugins,
-});
-
-app.listen(config.http_port, config.http_bind_address, () => {
-  console.log(`Listening on ${config.http_bind_address}:${config.http_port}...`);
-});
+startHeadless();
```

Comment:
Here we dispatch the async method without an await because the docker version of node does not support root level awaits, we can await this if we bump the version but this works as intended on all versions.